src/main.ts

```diff
@@ -49,12 +49,12 @@ export const main = async () => {
       // eslint-disable-next-line no-process-exit
       process.exit(1);
     }
+
     const {userFunction, signatureType} = loadedFunction;
-    const server = getServer(
-      userFunction!,
-      signatureType,
-      options.enableExecutionId
-    );
+    // It is possible to overwrite the configured signature type in code so we
+    // reset it here based on what we loaded.
+    options.signatureType = signatureType;
```

Comment:
The signature type is returned by `getUserFunction` and it may not match the signature type that was set in options. When that happens we want it take precedence, so we update the options to keep it consistent.